code/weapon/weapons.cpp

```diff
@@ -4588,10 +4588,6 @@ void weapon_update_missiles(object* obj, float  frame_time) {
 		if (wp->swarm_index >= 0) {
 			swarm_update_direction(obj);
 		}
-
-		if (wp->cscrew_index >= 0) {
-			cscrew_process_post(obj);
```

Comment:
How does is moving this function call out of `weapon_home` possible without affecting retail behavior and why not move `swarm_update_direction` as well?

Reply:
It's not terribly clear, but as I mentioned `cscrew_process_post` really is a form of post-processing, the missiles have already used `ai_turn_towards_vector` like regular missiles in `weapon_home` and the function basically just puts them in a 'fake' position to make it look like its corkscrewing when it really isn't. In order for corkscrew missiles to function properly they _must_ be in `obj_move_post`. This function is not related to AI or 'proper' turning at all. It was an oversight by me in #2613 to put them in `weapon_update_missiles`.

Unlike corkscrew missiles, swarm missiles will not use the `ai_turn_towards_vector` in `weapon_home`, `swarm_update_direction` uses it in a different way , and is _not_ a form of post-processing.